### ISSUE TITLE
2pass Error : Error in freed  returnVal 0

### ISSUE THREAD

**Issue by AlexBoswellVCD** [OpenVisualCloud/SVT-AV1#736]
**Error Description**
The encoder in Linux for the **second** pass will print "Error in freed  returnVal 0" and appear to finish encoding anyways
```
-------------------------------------------
SVT-AV1 Encoder
SVT [version]:	SVT-AV1 Encoder Lib v0.7.0
SVT [build]  :	GCC 7.4.0	 64 bit
LIB Build date: Oct 23 2019 18:59:26
-------------------------------------------
Number of logical cores available: 48
Number of PPCS 105
------------------------------------------- 
SVT [config]: Main Profile	Tier (auto)	Level (auto)	
SVT [config]: EncoderMode 							: 0 
SVT [config]: EncoderBitDepth / EncoderColorFormat / CompressedTenBitFormat	: 8 / 1 / 0
SVT [config]: SourceWidth / SourceHeight					: 640 / 360 
SVT [config]: Fps_Numerator / Fps_Denominator / Gop Size / IntraRefreshType 	: 25000 / 1000 / 120 / 1
SVT [config]: HierarchicalLevels / BaseLayerSwitchMode / PredStructure		: 4 / 0 / 2 
SVT [config]: BRC Mode / QP  / LookaheadDistance / SceneChange			: CQP / 20 / 33 / 0 
------------------------------------------- 
Encoding          Error in freed  returnVal 0
Error in freed  returnVal 0
Error in freed  returnVal 0
        1        2        3
SUMMARY --------------------------------- Channel 1  --------------------------------
Total Frames		Frame Rate		Byte Count		Bitrate
           3		25.00 fps		     75621		5041.40 kbps


Channel 1
Average Speed:		0.159 fps
Total Encoding Time:	18901 ms
Total Execution Time:	20330 ms
Average Latency:	16480 ms
Max Latency:		18900 ms
Encoder finished

```

**Reproducing the error**

I ran the following command on Linux for the first pass encode 

```
./SvtAv1EncApp -enc-mode 5 -w 640 -h 360 -n 3 -i blue_sky_360p_60f.yuv -q 20 -fps-num 25000 -fps-denom 1000 -intra-period 119 -output-stat-file a.stat -input-stat-file a.stat -bit-depth 8 -lad 119 -hdr 0 -enc-mode-2p 0
```

and then this following command to the second pass encode

```
./SvtAv1EncApp -enc-mode 0 -w 640 -h 360 -n 3 -i blue_sky_360p_60f.yuv -q 20 -fps-num 25000 -fps-denom 1000 -intra-period 119 -output-stat-file a.stat -input-stat-file a.stat -bit-depth 8 -lad 119 -hdr 0 
```

I confirmed the bug exists in the current master as well as 628437e

**Comment by luigi311** [comment]
Looks like you have output and input stat file on both passes. Based on the user guide i think you are only supposed to include the output-stat-file on the first pass and then the input-stat-file on the 2nd pass.

**Comment by luigi311** [comment]
Try
`SvtAv1EncApp -enc-mode 5 -w 640 -h 360 -n 3 -i blue_sky_360p_60f.yuv -q 20 -fps-num 25000 -fps-denom 1000 -intra-period 119 -output-stat-file Release_628437e_blue_sky_360p_60f_m50pass_Q20.stat -bit-depth 8 -lad 119 -hdr 0 -enc-mode-2p 0`
and
`SvtAv1EncApp -enc-mode 0 -w 640 -h 360 -n 3 -i blue_sky_360p_60f.yuv -q 20 -fps-num 25000 -fps-denom 1000 -b Release_628437e_M0_blue_sky_360p_60f__m50pass_Q20.bin -intra-period 119 -input-stat-file Release_628437e_blue_sky_360p_60f_m50pass_Q20.stat -bit-depth 8 -lad 119 -hdr 0`

**Comment by hassount** [comment]
@anaghdin 
interesting, should we shut the combination of the input / output file at the same time ?

**Comment by anaghdin** [comment]
Good idea. We also need to add some error checks on availability of the file.

**Comment by easyfab** [comment]
This 2pass commands aren't really user-friendly. 
Why not do like almost others encoders and using -pass option

-pass1 for first pass and -pass2 for the second, 
automatically generate and use a stat file (e.g av1_2pass.stat ) unless user define one  ( e.g -2pass_stat_file )

**Comment by luigi311** [comment]
@easyfab 
That is still to come. It looks like they want to implement a --passes N flag on your command and it will auto run through the N commands it needs to do to go through the multiple different passes. There is no estimate on when it will be implemented as of yet though just that it is a good idea so it is more user friendly.

https://github.com/OpenVisualCloud/SVT-AV1/pull/707#issuecomment-543404292

**Comment by easyfab** [comment]
@luigi311 
a -passes N flag could indeed be a good option but a simple -pass1/-pass2 flag also when needed ( e.g. using pipe input )

**Comment by luigi311** [comment]
@easyfab Yup they said they were going to go with both options just in case.

**Comment by hassount** [comment]
@JackZhouVCD Can we re-verify this ?

**Comment by JackZhouVCD** [comment]
> @JackZhouVCD Can we re-verify this ?

Yes this is still seen in 3ea18c2

**Comment by ccccheung** [comment]
Can still reproduce this bug on Linux and windows  with the command line suggested in the original issue as well as the one that was suggested in the comments in https://github.com/OpenVisualCloud/SVT-AV1/commit/326f320be02ef0d78f6520fa2835b29624b28955

**Comment by hassount** [comment]
Let's re-check this soon after the 2-pass work gets merged

**Comment by hassount** [comment]
@anaghdin is this addressed with #1431 ?

**Comment by anaghdin** [comment]
yes it does

> @anaghdin is this addressed with #1431 ?

**Comment by hassount** [comment]
Great, thank you
